src/MembershipManager.php

```diff
@@ -11,16 +14,16 @@
 use Drupal\og\Entity\OgMembership;
 
 /**
- * Membership manager.
+ * Service for managing memberships and group content.
  */
 class MembershipManager implements MembershipManagerInterface {
 
   /**
-   * Static cache of the memberships and group association.
+   * The static cache backend.
    *
-   * @var array
+   * @var \Drupal\Core\Cache\CacheBackendInterface
    */
-  protected $cache;
+  protected $staticCache;
```

Comment:
@pfrenssen, I wonder why we're only doing memory caching here. This looks for me a good candidate to double the memory caching with persistent caching (database, Redis, etc.). All we need here is to declare a `\Drupal\Core\Cache\BackendChain` service and add a memory and a cache bin backends. That would improve dramatically the performance.

Reply:
I've opened #485 to fix this.